js/content/community.js

```diff
@@ -291,6 +292,260 @@ let CommentHandler = (function(){
+                    case URLPath.includes("/status/"): // new comment in status
+                    case URLPath.includes("/friendactivitydetail/"): // new comment in purchase
+                        return false;
+                        break;
+                    case URLPath.includes("/recommended/"): // new comment in review, review edit; no break
+                        return true;
+                        break;
+                    case !!URLPath.match(/\/(id|profiles|groups)\/[^\/]+\/?$/): // new comment in profile/group
+                    case URLPath.includes("/sharedfiles/filedetails/"): // new comment in screenshot/artwork/Workshop/Greenlight
+                    case document.location.href.includes("announcements/detail/"): // new comment in announcement of a group/game; may be either path or hash
+                    case URLPath.includes("/news/"): // new comment in news; Store & Community has separate news
+                    //case URLHash.includes("events/"): // new comment in group event; falls under group path match
+                    //case URLHash.includes("comments"): // new comment on group's comments page; falls under group path match
+                    case URLPath.includes("/allcomments"): // new comment on profile's comments page
+                        return false;
+                        break;
+                    case !!URLPath.match(/\/discussions(\/forum)?\/\d{1,2}\/\d+/): // new comment, comment edit on forum; no break
+                    case URLPath.includes("/reporteddiscussions/"): // new comment in report
+                        return true;
+                        break;
+                    case URLPath.includes("/discussions"): // new topic, topic edit on forum; report edit; trailing slash omitted to work with group forum index
+                        return true;
+                        break;
+                    case document.location.href.includes("store.steampowered.com/app/"): // new review
+                        return true;
+                        break;
+                    case !!URLPath.match(/\/announcements\/(create|edit)/): // new group announcement
+                        return true;
+                        break;
+                    case !!URLPath.match(/\/edit(\/profile)?$/): // user/group profile edit
+                        return false;
+                        break;
+                    case URLPath.includes("/sharedfiles/itemedittext/"): // screenshot/artwork/workshop edit
+                        return true;
+                        break;
+                    case URLPath.includes("/sharedfiles/edititem/"): // new artwork
+                        return true;
+                        break;
+                    default: // kinda default
+                        return false;
+                        break;
+                }
+            }
+
+            BBCode_SpoilerSelection = function() {
```

Comment:
My personal style for this would pack the `BBCode_` namespace into a collection and have the code, localization key and icon configured together, then build the toolbar from the configuration instead of a HTML string with the icon &c.. embedded into it.

Reply:
It's a bit too complex to do it like that. They are all a bit too diverse. Some use native steam code, other don't. Some use images, some use text. Some have different style. Etc...